File: Documentação/Documentação/Template Latex - Relatorio Tecnico - IFSP - SBV/fontes/Raylib/spawnPiece.cpp
void Tetris::spawnPiece(){
    if (piece != nullptr) {
        delete piece;
        piece = nullptr;
    }

    Vector2 spawnPosition = {4,0};
    if(bagPieces.empty()){
        refillBag();
    }
    
    int num = bagPieces.back();
    bagPieces.pop_back();

    Piece newPiece(this, num, spawnPosition);
    if(isSpawnBlocked(&newPiece, spawnPosition)){
        gameover = true;
        piece = nullptr;
    } else {
            piece = new Piece(this, num, spawnPosition);
    }
}

void Tetris::refillBag(){
    bagPieces.push_back(1);
    bagPieces.push_back(2);
    bagPieces.push_back(3);
    bagPieces.push_back(4);
    bagPieces.push_back(5);
    bagPieces.push_back(6);
    bagPieces.push_back(7);
    shuffleBag();
}

void Tetris::shuffleBag(){
    std::shuffle(bagPieces.begin(), bagPieces.end(), generator);
}

File: Documentação/Documentação/Template Latex - Relatorio Tecnico - IFSP - SBV/fontes/Pygame/spawnPiece.py
def spawnPiece(self):
        nextPiece = Piece(self.getNextPiece(), Vector2(5,1))
        if self.isSpawnBlocked(nextPiece):
            self.gameOver()
            return
        else:
            self.piece = nextPiece

def getNextPiece(self):
    if len(self.bag) == 0:
        self.refillBag()
    return self.bag.pop()

def refillBag(self):
    self.bag = [1,2,3,4,5,6,7]
    random.shuffle(self.bag)